hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -512,6 +531,11 @@ public int preferredLocalParallelism() {
             return 1;
         }
 
+        @Override
+        public boolean isReusable() {
+            return true;
+        }
```

Comment:
This one looks safe because it doesn't do anything in init, but what if the member with `memberAddress` leaves after the plan is cached?

Reply:
It looks like `get()` covers edge cases.

Reply:
Right, somehow I have missed that. And if a job ends with an exception, is the plan invalidated? Otherwise, the query could be failing until the cache is cleared.

Reply:
In SQL, `PlanCache` caches `SqlPlan`s, which are just parsed versions of SQL statements. They are basically ASTs, not execution plans. In Jet, `ExecutionPlan`s are not cached.

Reply:
`PlanCache` contains `DAG` for some statements (see `SelectPlan` or `DmlPlan`), for other rel nodes it contains different things.

Reply:
> member with memberAddress leaves after the plan is cached?

This is a good question. SQL uses only local member address with this PMS (`forceTotalParallelismOne`) and plan cache is not distributed. So the member will leave together with the plan. But using any other member explicitly is asking for trouble and such plan would become permanently invalid.